src/buttons/EditButton/EditButton.tsx

```diff
@@ -1,41 +1,59 @@
 import EditIcon from '@mui/icons-material/Edit';
+import { ListItemIcon, MenuItem } from '@mui/material';
 import IconButton, { IconButtonProps } from '@mui/material/IconButton';
 import Tooltip from '@mui/material/Tooltip';
 
 import React, { FC, MouseEventHandler } from 'react';
 
+import { ActionButton, ActionButtonVariant } from '../../types';
+
 export type Props = {
   id?: string;
   className?: string;
   ariaLabel?: string;
   onClick?: MouseEventHandler;
   tooltip?: string;
   size?: IconButtonProps['size'];
+  type?: ActionButtonVariant;
 };
 
 const EditButton: FC<Props> = ({
   id,
   className,
   ariaLabel,
   onClick,
-  tooltip = 'edit',
+  tooltip = 'Edit',
   size = 'small',
+  type = 'icon',
 }) => {
-  return (
-    <Tooltip title={tooltip}>
-      <span>
-        <IconButton
-          id={id}
-          aria-label={ariaLabel}
-          className={className}
-          onClick={onClick}
-          size={size}
-        >
-          <EditIcon />
-        </IconButton>
-      </span>
-    </Tooltip>
-  );
+  switch (type) {
+    case ActionButton.MENU_ITEM:
+      return (
+        <MenuItem key={tooltip} onClick={onClick} id={id} className={className}>
```

Comment:
Menus have a `dense` prop which makes them smaller, maybe we could have something like `dense={size === "small"}` to be consistent with the size prop ? Currently there is no change when the button is in "menu" mode.